applications/MeshingApplication/custom_utilities/tetrahedra10_mesh_converter_utility.h

```diff
@@ -0,0 +1,356 @@
+    * @param New_Elements: The new elements created
+    * @param interpolate_internal_variables: A boolean that defines if it is necessary to interpolate the internal variables
+    * @return this_model_part: The model part of the model (it is the input too)
+    */
+
+    void EraseOldElementAndCreateNewElement(
+            ModelPart& this_model_part,
+            const compressed_matrix<int>& Coord,
+            PointerVector< Element >& NewElements,
+            bool interpolate_internal_variables
+    ) override
+    {
+	ElementsArrayType& rElements = this_model_part.Elements();
+	ElementsArrayType::iterator it_begin = rElements.ptr_begin();
+	ElementsArrayType::iterator it_end = rElements.ptr_end();
+
+	const ProcessInfo& rCurrentProcessInfo = this_model_part.GetProcessInfo();
+	int edge_ids[6];
+	std::vector<int> node_ids;
+
+	for (ElementsArrayType::iterator it = it_begin; it != it_end; ++it)
+	{
+        GlobalPointersVector< Element >& rChildElements = it->GetValue(NEIGHBOUR_ELEMENTS);
+        rChildElements.resize(0);
+
+        CalculateEdges(it->GetGeometry(), Coord, edge_ids, node_ids);
+
+        // Generate the new Tetrahedra3D10 element
+        Tetrahedra3D10<Node<3>> geom = GenerateTetrahedra(this_model_part, node_ids);
+        const Element& rElem = KratosComponents<Element>::Get("Element3D10N");
+        Element::Pointer p_element;
+        p_element = rElem.Create(it->Id(), geom, it->pGetProperties());
+        p_element->Initialize(rCurrentProcessInfo);
+        p_element->InitializeSolutionStep(rCurrentProcessInfo);
+        p_element->FinalizeSolutionStep(rCurrentProcessInfo);
+
+        // Setting the internal variables in the "child" elem (the element replacing the old one)
+        if (interpolate_internal_variables == true)
+        {
+            //This method only copies the current information to the new element
+            InterpolateInteralVariables(0, *it.base(), p_element, rCurrentProcessInfo);
+        }
+
+        // Transfer elemental variables to new element
+        p_element->Data() = it->Data();
+        p_element->GetValue(SPLIT_ELEMENT) = false;
+        NewElements.push_back(p_element);
+
+        rChildElements.push_back( Element::WeakPointer(p_element) );
+    }
+
+    // Now replace the elements in SubModelParts
+    if ( NewElements.size() > 0 ) ReplaceElementsInSubModelPart(this_model_part);
+}
+
+    /**
+    * It replaces the old Tetrahedra3D4 elements in its corresponding submodelpart by its new Tetrahedra3D10
+    * @param this_model_part: The modelpart or submodelpart to replace the elements
+    */
+    void ReplaceElementsInSubModelPart(ModelPart& this_model_part) {
+        for(Element::Pointer& p_element : this_model_part.ElementsArray()){
+                if( p_element->GetValue(SPLIT_ELEMENT) )
+                {
+                    GlobalPointersVector< Element >& children = p_element->GetValue(NEIGHBOUR_ELEMENTS);
```

Comment:
```suggestion
                    auto& r_children = p_element->GetValue(NEIGHBOUR_ELEMENTS);
```

Reply:
I would leave it this way since it think helps to visualize what we are dealing with. Personally, when reading some similar code for the first time, I was really thankful that this code was explicit, so I would prefer to leave it this way. 😄

Reply:
Totally agree. My point goes in the direction of code maintenance. Imagine that in the future (because reasons) we change the `NEIGHBOUR_ELEMENTS` type, then we'll need to go one by one changing the declarations. Having said so, leave it as you want 😉 .